.github/workflows/gradle.yml

```diff
@@ -9,30 +9,30 @@ on:
 
 jobs:
   build:
-
     runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
-        java-version: [
-            8,
-            11,
-            17,
-        ]
-        os: [
-          windows-latest,
-          ubuntu-latest,
-        ]
+        java-version:
+          - 8
+          - 11
+          - 17
+        os:
+          - windows-latest
+          - ubuntu-latest
+          - macos-latest
       fail-fast: false
 
     name: Java ${{ matrix.java-version }} on ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up JDK ${{ matrix.java-version }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
+          distribution: 'temurin'
```

Comment:
Please add a comment on why we need this particular one.

Reply:
I added a comment. With the latest setup-java action, you need to specify a distribution, and Temurin seems to be suggested by a few sites I checked, but there is no "definite right answer"...